tests/unit/gtArgsFilter.cpp

```diff
@@ -0,0 +1,392 @@
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+/*
+ * gtArgsFilter.cpp
+ */
+
+#include "argsfilter.h"
+#include "common_test.h"
+#include "gtest/gtest.h"
+using namespace common_test;
+
+class argsfilter : public BaseFixture, public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+    m_Properties = NULL;
+    EXPECT_EQ(FPGA_OK, fpgaGetProperties(NULL, &m_Properties));
+  }
+
+  virtual void TearDown() {
+    EXPECT_EQ(FPGA_OK, fpgaDestroyProperties(&m_Properties));
+  }
+
+  fpga_properties m_Properties;
+};
+
+/**
+ * @test       none
+ *
+ * @brief      Given an argument vector with no options related to
+ *             bus, device, function or socket id <br>
+ *             When I call fpgaPropertiesFromArgs with that argument vector<br>
+ *             Then the return value is 0, the fpga_result object is
+ *             unchanged, and my argument vector remains unchanged as well
+ *
+ */
+TEST_F(argsfilter, none) {
+  fpga_result res = FPGA_EXCEPTION;
+  int argc = 5;
+  const char *argv[5] = {"prog", "--opt1", "val1", "word1", "word2"};
+  int exit_code = fpgaPropertiesFromArgs(m_Properties, &res, &argc,
+                                         const_cast<char **>(argv));
+  EXPECT_EQ(exit_code, 0);
+  EXPECT_EQ(res, FPGA_EXCEPTION);
+  EXPECT_EQ(5, argc);
+  EXPECT_STREQ(argv[0], "prog");
+  EXPECT_STREQ(argv[1], "--opt1");
+  EXPECT_STREQ(argv[2], "val1");
+  EXPECT_STREQ(argv[3], "word1");
+  EXPECT_STREQ(argv[4], "word2");
+}
+
+/**
+ * @test       bus_long
+ *
+ * @brief      Given an argument vector with --bus options set to '0x5e'<br>
+ *             When I call fpgaPropertiesFromArgs with that argument vector<br>
+ *             Then the return value is 0, the fpga_result object is<br>
+ *             FPGA_OK, and my argument vector is changed so that --bus and
+ *             0x5e are removed.
+ *
+ */
+TEST_F(argsfilter, bus_long) {
+  fpga_result res = FPGA_EXCEPTION;
+  int argc = 7;
+  uint8_t value = 0;
+  const char *argv[7] = {"prog", "--opt1", "val1", "--bus",
+                         "0x5e", "word1",  "word2"};
+  int exit_code = fpgaPropertiesFromArgs(m_Properties, &res, &argc,
+                                         const_cast<char **>(argv));
+  EXPECT_EQ(FPGA_OK, fpgaPropertiesGetBus(m_Properties, &value));
+  EXPECT_EQ(0x5e, value);
+  EXPECT_EQ(exit_code, 0);
+  EXPECT_EQ(res, FPGA_OK);
+  EXPECT_EQ(5, argc);
+  EXPECT_STREQ(argv[0], "prog");
+  EXPECT_STREQ(argv[1], "--opt1");
+  EXPECT_STREQ(argv[2], "val1");
+  EXPECT_STREQ(argv[3], "word1");
+  EXPECT_STREQ(argv[4], "word2");
```

Comment:
I don't follow what this is doing.  Does fpgaPropertiesFromArgs() mutate the argv parameter?

Reply:
Yes, as mentioned in the note in the description of this PR. 😄

Reply:
Doh.. (slurps coffee to improve focus)